packetnetworking/utils.py

```diff
@@ -301,39 +296,47 @@ def get_lshw_info():
     return json.loads(get_output(["lshw", "-json", "-disable", "fb"]).decode())
 
 
+def get_devices_info():
+    path = "/sys/class/net/"
+    nics = []
+    ifaces = get_output(["ls", path]).decode().split()
+    for iface in ifaces:
+        try:
+            driver_path = (
+                get_output(["readlink", path + iface + "/device/driver"])
+                .decode()
+                .strip()
+            )
+            driver = get_output(["basename", driver_path]).decode().strip()
+        except subprocess.CalledProcessError:
+            continue
+
+        serial = get_output(["cat", path + iface + "/address"]).decode().strip()
+
+        n = {"logicalname": iface, "serial": serial, "driver": driver}
```

Comment:
serial? this is the mac address not a serial address right?

Reply:
yeah it is the mac. I was just going off of the existing code that `nic["serial"]` is the mac: https://github.com/packethost/packet-networking/blob/master/packetnetworking/utils.py#L257

should I also change that part as well to `nic["mac"]` so it's not confusing?

Reply:
oh, it's because originally lshw lists the mac under "serial", that's why the existing code was the way it is